README.md

```diff
@@ -215,6 +215,27 @@ The schema for this type is as follows:
 }
 ```
 
+### Generating an external manifest
+
+The `-s` option puts the manifest in an external sidecar file in the same location as the output file. The
+manifest will have the same output filename but with a ".c2pa" extension. The output file will be untouched. 
+
+```shell
+c2patool -s sample/test.json -o new_manifest.jpg
+```
+
+### Generating a remote manifest
+
+The `-r` option places an http reference to manifest in the output file. The manifest is returned as an external sidecar file in the same location as the output file. The manifest will have the same output filename but with a ".c2pa" extension. The manifest should then be placed at the location specified by the `-r` option. When using remote manifests the remote URL should be publicly accessible to be most useful to users. When verifying an asset, remote manifests are automatically fetched. You can disable this by removing feature `fetch_remote_manifests` from Cargo.toml.
+
+```shell
+c2patool -r http://my_server/myasset.c2pa sample/test.json -o new_manifest.jpg
+```
+
+In the example above c2patool will try to fetch the manifest for new_manifest.jpg from http://my_server/myasset.c2pa during validation.
+
+Note: It is possible to combine the `-s` and `-r` options. When used together a manifest will be embedded in the output files and the remote reference will also be added. The sidecar .c2pa file is omitted.
```

Comment:
Wait. Using `-r` _changes_ the meaning of `-s` so that the manifest is _embedded_? That seems kind of wrong to me.

Reply:
It actually generate the sidecar as described but Gavin pointed out that he did not need/want it in this case so I don't copy it to the output folder.  I am ambivalent about this feature and it is OK with me if the c2patool does not support this use case.

Reply:
@gpeacock Do you need this feature in c2patool?

Reply:
I need to think about this. The command line options for c2patool are tricky to get right.
This is why I'm planning to change the way the command line options work.
-s seems strange because it might copy but won't modify the output file.
-r is almost like -s except it will inject the remote url into the output file.
combining -s and -r is trying to handle the "both" case (embedded + remote) but it isn't actually like either of them. It seems too confusing. I suggest we leave that out for now. 
The cleanest most orthogonal way to handle this would be three separate flags, adding a -x flag to indicate "don't embed". That way -s always generates a sidecar, -r only embeds a specified url, and -x means don't embed c2pa data in the file.

-r  embeds a manifest and a remote url
-s creates a sidecar and embeds a manifest without adding a remote url
-x do not embed a manifest (could be used on its own to preview a manifest)
-s -x means create a sidecar with nothing embedded.

-s -r creates a sidecar and embeds both the c2pa manifest and a remote url

-x -r would just inject a remote url but not create a sidecar or embedded manifest

Reply:
You are forgetting one important thing I cannot validate, which you use to dump, without the source asset.  That is why output always generates the output asset.  The sidecar may or may not need to exist.

Reply:
If you want to make it work the way you describe you will need a new way to validate.

Reply:
Another, maybe simpler way to parse this is:
-r just embeds a remote url -it does not generate a sidecar on its own.
-s creates a sidecar and does not embed a c2pa manifest.
if both are used together you get a sidecar and an embedded remote url, but no embedded manifest.

i.e:  -s always means an external manifest with nothing embedded, and -r always means an embedded url

That way you can still create an embedded manifest with a remote url if you want it.
I think i like this the best for now.